src/backend/distributed/citus--8.3-1--8.4-1.sql

```diff
@@ -0,0 +1,95 @@
+/* citus--8.3-1--8.4-1 */
+
+/* bump version to 8.4-1 */
+CREATE OR REPLACE FUNCTION pg_catalog.citus_prepare_pg_upgrade()
+    RETURNS void
+    LANGUAGE plpgsql
+    SET search_path = pg_catalog
+    AS $cppu$
+BEGIN
+    -- backup citus catalog tables
+    CREATE TABLE public.pg_dist_authinfo AS SELECT * FROM pg_catalog.pg_dist_authinfo;
```

Comment:
To make this work on community version as well, shall we add another function, `citus_enterprise_prepare_pg_upgrade()`, which is a no-op here, but contains the accesses to enterprise-only metadata tables on the same file for citus-enterprise repo? And, call `citus_enterprise_prepare_pg_upgrade()` at the end of this function, remove pg_dist_authinfo/ pg_dist_poolinfo from this function.

Reply:
`pg_dist_authinfo` and `pg_dist_poolinfo` are already available in community, only used in enterprise.

IIRC there is no SQL difference between community and enterprise. The only difference is the binary. Meaning, if we only want to restore these catalogs in enterprise we need to migrate the logics to C code. Until then we need to have the backup and restore of those tables here I guess.